js/app.js

```diff
@@ -117,382 +84,155 @@ app.controller('CheckoutController', function($scope, $interval, Order, $httpPar
+                //Wait for 2 seconds for order status to update on server
+                }, 2000, 1);
             }
-            return false;
         }
     }
-
-    //Add a refund address to altcoin order
-    $scope.add_refund_click = function() {
-        $scope.altrefund = document.getElementById("bnomics-refund-input").value;
-        $scope.altuuid = get_uuid();
-        var response = AltcoinAddRefund.save({
-                'uuid': $scope.altuuid,
-                'address': $scope.altrefund
-            },function successCallback(data) {
-                if(data.result == 'ok') {
-                    send_refund_email();
-                    update_altcoin_status('refunded');
-                }else if(data.errors){
-                    var refund_message = document.getElementById("bnomics-refund-errors");
-                    refund_message.innerHTML = "";
-                    for (var key in data.errors){
-                        var value = data.errors[key];
-                        for (var i = value.length - 1; i >= 0; i--) {
-                            refund_message.innerHTML += "<p style='color:red'>"+value[i]+"</p>";
-                        }
-                    } 
+    
+    //Check if the blockonomics uuid is present
+    function check_blockonomics_uuid() {
+        $scope.spinner = true;
+        if (typeof $scope.order_uuid != 'undefined') {
+            //Fetch the order using uuid
+            Order.get({
+                "get_order": $scope.order_uuid,
+                "blockonomics_currency": $scope.currency.code
+            }, function(data) {
+                $scope.spinner = false;
+                if(data.txid !== undefined && data.txid !== ""){
+                    $scope.txid = data.txid;
+                    $scope.pending_error = true;
+                }else if(data.addr !== undefined){
+                    proccess_order_data(data);
+                    $scope.checkout_panel  = true;
+                }else if($scope.currency.code === 'btc'){
+                    $scope.address_error_btc = true;
+                }else if($scope.currency.code === 'bch'){
+                    $scope.address_error_bch = true;
                 }
             });
+        }
     }
-
-    //Go to add refund page
-    $scope.get_refund = function() {
-        $scope.hide_refund_reason = true;
-        update_altcoin_status('add_refund');
+    
+    $scope.spinner = true;
+    if(Object.keys($scope.active_currencies).length === 1){
+        // Auto select btc if 1 activated currency
+        $scope.currency = $scope.active_currencies['btc'];
+        $scope.currency.code = 'btc';
+        check_blockonomics_uuid();
+    }else if(Object.keys($scope.active_currencies).length >= 1){
+        //Show user currency selector if > 1 activated currency
+        $scope.currency_selecter  = true;
+        $scope.spinner = false;
+    }
+    
+    //Copy bitcoin address to clipboard
+    $scope.blockonomics_address_click = function() {
```

Comment:
This and `blockonomics_amount_click()` have the same code except element id. Maybe pass `divid` as argument for the function

Reply:
Updated in b6ffe22